source/PlayState.hx

```diff
@@ -1,16 +1,187 @@
 package;
 
+import flixel.text.FlxText;
+import entities.collectibles.Coin;
+import entities.terrain.Lava;
+import entities.projectiles.Fireball;
+import flixel.FlxObject;
+import flixel.FlxG;
+import flixel.group.FlxGroup.FlxTypedGroup;
+import entities.player.Hero;
+import entities.terrain.Wall;
 import flixel.FlxState;
 
 class PlayState extends FlxState
 {
-	override public function create()
+
+	private static var WORLD_LEFT(default,never) = -(1280*2);
+	private static var WORLD_RIGHT(default,never) = 1280*2;
+	private static var WORLD_TOP(default,never) = -(960*2);
+	private static var WORLD_BOTTOM(default,never) = 960*2;
+
+	private static var WALL_COUNT(default, never) = 10;
+	private static var WALL_START_X(default, never) = 150;
+	private static var WALL_START_Y(default, never) = 200;
+
+	private static var LAVA_COUNT(default, never) = 10;
+	private static var LAVA_START_X(default, never) = 150;
+	private static var LAVA_START_Y(default, never) = 232;
+
+	private static var FIREBALL_COUNT(default, never) = 10;
+	private static var FIREBALL_SPAWN_BORDER(default, never) = 50;
+
+	private var hero:Hero;
+	private var coin:Coin;
+	private var walls:FlxTypedGroup<Wall>;
+	private var fireballs:FlxTypedGroup<Fireball>;
+	private var lavaPools:FlxTypedGroup<Lava>;
+
+	var score:FlxText;
+
+	override public function create():Void
 	{
 		super.create();
+
+		hero = new Hero();
+		add(hero);
+		FlxG.camera.target = hero;
+		FlxG.camera.zoom = .5;
+		initializeWalls();
+		initializeLava();
+		initializeFireballs();
+		coin = new Coin();
+		add(coin);
+		score = new FlxText(100,100,0,"0",32);
+		add(score);
+	}
+
+	private function initializeWalls() {
+		FlxG.worldBounds.set(WORLD_LEFT-32,WORLD_TOP-32,WORLD_RIGHT-WORLD_LEFT+64,WORLD_BOTTOM-WORLD_TOP+64);
```

Comment:
General note: if you're wanting to do logic or set up positioning based on the size of your "wall" objects, then it's best to make these `32`s instead be some constant variable defined for wall size. `Wall.WIDTH` or something like that. 

This would save you a massive headache if you ever wanted to come back and touch this code again (and also makes it easer to understand what all of these `32`s are supposed to represent).